runtime/calamari/src/lib.rs

```diff
@@ -571,6 +575,23 @@ impl pallet_scheduler::Config for Runtime {
 	type MaxScheduledPerBlock = MaxScheduledPerBlock;
 	type WeightInfo = weights::pallet_scheduler::SubstrateWeight<Runtime>;
 	type OriginPrivilegeCmp = OriginPrivilegeCmp;
+	type PreimageProvider = Preimage;
+	type NoPreimagePostponement = NoPreimagePostponement;
+}
+
+parameter_types! {
+	pub const PreimageMaxSize: u32 = 4096 * 1024;
+	pub const PreimageBaseDeposit: Balance = 1 * KMA;
```

Comment:
What does this constant mean?

Reply:
This is an amount which gets reserved every time you submit a preimage.
It will get unreserved when the preimage is requested, i.e. when it is going to be used.

Reply:
Another arbitrary choice. Maybe ours is too small.
Karura is around ~6 KAR
Interlay is around ~6itbc
KILT is 1 KILT

Reply:
No, I don't think so. Someone can also upload a 4MB remark.
The tx fee is separate as well.

Reply:
IMO we can set `PreimageBaseDeposit` between 1k KMA and 10k KMA , so it's more consistent with other networks. But this is still kind of arbitrary choice.

Reply:
So, this is signed extrinsic? As a result, the one who submit preimage still need to pay gas fee? 

Also, 4MB is not a reasonable setting in my view. This reason is our current block size and normal_dispatch ratio setting:

https://github.com/Manta-Network/Manta/blob/manta/runtime/calamari/src/lib.rs#L134

A reasonable size limit is 3.5 MB unless we change the normal_dispatch ratio setting. 

I tend to like to merge this PR as it is, but could you log these 2 issues and put it to next release?

Reply:
Aaah this is where the 4MB comes from - 80% of the 5MB limit in Polkadot, good catch.

Submitting the preimage is a signed extrinsic. The actual check is `ensure_signed_or_manager` where the Manager is configurable. In our case - `type ManagerOrigin = EnsureRoot<AccountId>;`
Let's merge this, I've included these two issues in the 3.1.5 milestone:
https://github.com/Manta-Network/Manta/issues/386
https://github.com/Manta-Network/Manta/issues/400

Reply:
There is a bit room for that. This is not super strict. But I think it is still a good practice to control this under the normal_dispatch_ratio.